pteranodon/autoconnect.py

```diff
@@ -0,0 +1,234 @@
+import asyncio
+from typing import List, Optional, Tuple, Any, Dict
+import os
+import re
+import socket
+
+from pymavlink import mavutil
+
+
+# TODO_: Better name?
+class AutoConnect:
+    """
+    Searches serial devices and, if a server ip address is provided, that server for open MAV SDK servers.
+    """
+
+    def __init__(self, server_ip_addr: Optional[str]):
+        """
+        Searches serial devices and, if a server ip address is provided, that server for open MAV SDK servers.
+
+        :param server_ip_addr: The server IP address to search for MAV SDK searches, None if only serial devices should be searched.
+        :type server_ip_addr: Optional[str]
+        """
+        self._server_addr = server_ip_addr
+        self._should_check_server = self._server_addr is not None
```

Comment:
We should default to localhost or an empty string if nothing is given. When connecting to a mavsdk server we use "udp://:14540" to denote localhost. If nothing is provided it does not mean serial